### PR TITLE
Auth review

### PR DESCRIPTION
**Pull request by oluwoleilesanmi**
Kindly help review.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,28 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
-  include SessionsHelper
+  helper_method :current_user
```
Comment (heyapricot) [inline app/controllers/application_controller.rb]: With this line you are sharing the `current_user` method to the viewers and we are back at the original problem

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,28 @@
...
+      if user && user.authenticated?(cookies.signed[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end 
+
+  def log_in(user)
+    session[:user_id] = user.id
+  end
```
Comment (heyapricot) [inline app/controllers/application_controller.rb]: Do all controllers need to inherit the `log_in` method from the Application Controller?

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,4 +1,28 @@
...
+  
+  def require_user 
+    redirect_to '/login' unless current_user 
+  end
```
Comment (heyapricot) [inline app/controllers/application_controller.rb]: Do all controllers need to inherit this method from the Application Controller?

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -17,6 +18,10 @@ def index
     @posts = Post.all
   end
 
+  def self.first_name(id)
+    User.first_name(id)
+  end
```
Comment (heyapricot) [inline app/controllers/posts_controller.rb]: Why did you build this method?

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,21 +1,27 @@
...
-    redirect_to '/' 
-  else 
-    redirect_to '/signup' 
-  end 
+ @user = User.new(user_params) 
+ if @user.save 
+   session[:user_id] = @user.id 
+   @is_new_account = true
+   render :index
+   @is_new_account = false
```
Comment (heyapricot) [inline app/controllers/users_controller.rb]: What is the purpose of the `@is_new_account` variable?

File: app/models/user.rb [hunk 6]
```diff
@@ -1,8 +1,10 @@
 class User < ApplicationRecord
-  has_many :photos, dependent: :destroy
+  has_many :posts, dependent: :destroy
   has_secure_password
   attr_accessor :remember_token
 
+  scope :first_name, ->(user_id) { where(id: user_id).pluck(:first_name) }
+
```
Comment (heyapricot) [inline app/models/user.rb]: What's the purpose of this?

File: app/views/posts/index.html.erb [hunk 7]
```diff
@@ -5,6 +5,7 @@
         <div>
           <% @posts.each do |post| %>
             <ul>
+              <li><span>event by </span><%=PostsController.first_name(post.user_id) %> <br></li>
```
Comment (heyapricot) [inline app/views/posts/index.html.erb]: This is overkill. Why did you build a custom method? Couldn't you have done this just with the associations?

Comment (oluwoleilesanmi) [reply app/views/posts/index.html.erb]: It depends on how you want to see things, there is really no end to refactoring.
The reason is do you think the view should be calling the database directly ?.

File: app/models/user.rb (hunk 6 above)
Comment (oluwoleilesanmi) [reply app/models/user.rb]: That is a database query or scope to get the creators of a post.

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (oluwoleilesanmi) [reply app/controllers/users_controller.rb]: I tried to solve a problem with the sequencing of the ui's displayed, and that was the solution.

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (oluwoleilesanmi) [reply app/controllers/posts_controller.rb]: It depends on how you want to see things, there is really no end to refactoring.
The reason is, do you think the view should be calling the database directly ?

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (oluwoleilesanmi) [reply app/controllers/application_controller.rb]: No but there is a dependency between log_in and current_user.

**Pull request closed without merging**